### PR TITLE
- ec2utils

### PR DESCRIPTION
**Pull request by rjschwei**
+ Keep security group id specification as optional
    ~ Read from config file
    ~ Create temporary sg in proper VPC is not specified

### CODE REVIEW

File: ec2utils/ec2uploadimg/ec2uploadimg [hunk 1]
```diff
@@ -519,7 +527,40 @@ try:
...
+                    try:
+                        security_group_ids = utils.get_from_config(
+                            args.accountName,
+                            config,
+                            region,
+                            'security_group_ids_%s' % region,
+                            '--security-group-ids'
+                        )
+                        if args.verbose and security_group_ids:
+                            msg = 'Using Seccurity Group IDs: %s'
```
Comment (smarlowucf) [inline ec2utils/ec2uploadimg/ec2uploadimg]: s/Seccurity/Security

File: ec2utils/ec2uploadimg/ec2uploadimg [hunk 2]
```diff
@@ -519,7 +527,40 @@ try:
...
+                        ec2 = boto3.client(
+                            aws_access_key_id=access_key,
+                            aws_secret_access_key=secret_key,
+                            region_name=region,
+                            service_name='ec2'
+                        )
+                        subnet_data = ec2.describe_subnets(
+                            SubnetIds=[vpc_subnet_id]
+                        )
+                        vpc_id = subnet_data['Subnets'][0]['VpcId']
```
Comment (smarlowucf) [inline ec2utils/ec2uploadimg/ec2uploadimg]: This assumes the subnet exists. Maybe good to catch IndexError and handle accordingly.

**Review by smarlowucf — CHANGES_REQUESTED** [review]

File: ec2utils/ec2uploadimg/ec2uploadimg [hunk 3]
```diff
@@ -442,11 +446,10 @@ try:
 
             key_pair_name = args.sshName
             ssh_private_key_file = args.privateKey
-            if (
-                not key_pair_name
-                and not ssh_private_key_file
-                and not args.accountName
-               ):
+            if (not key_pair_name and not
```
Comment (bear454) [inline ec2utils/ec2uploadimg/ec2uploadimg]: Is this a necessary change? IMO the old format was more readable.

**Review by bear454 — APPROVED** [review]

File: ec2utils/ec2uploadimg/ec2uploadimg (hunk 2 above)
Comment (bear454) [reply ec2utils/ec2uploadimg/ec2uploadimg]: I'm good either way; if it doesn't exist, the error _should_ bubble up.

**Comment by rjschwei** [comment]
@smarlowucf better?

**Review by smarlowucf — APPROVED** [review]

**Comment by smarlowucf** [comment]
:+1:

**Pull request merged**